Dockerfile

```diff
@@ -22,28 +25,39 @@ COPY . .
 # Compile with optimizations
 RUN swift build --enable-test-discovery -c release
 
+# Switch to the staging area
+WORKDIR /staging
+
+# Copy main executable to staging area
+RUN cp "$(swift build --package-path /build -c release --show-bin-path)/Run" ./
+
+# Uncomment the next line if you need to load resources from the `Public` directory.
+#RUN mv Public /staging/Public
```

Comment:
Could we make this a template option? (In a different PR)

Reply:
Maybe, but then that begs the question of including Leaf as a template option as well. This PR needs @tanner0101's eye anyway - this is definitely a question I'd like his input on. Tanner?

Reply:
Yeah, agree Leaf should be a part of the template too.

Reply:
This has been discussed in #32 and I think there's a general consensus to add it

Reply:
Awesome. Thanks @0xTim!

Reply:
I agree with the above - I think copying `Resources/Views` as a commented out line should be added in as well

Reply:
The template includes a commented out line for enabling FileMiddleware: https://github.com/vapor/template/blob/master/Sources/App/configure.swift#L8. So having this line in the dockerfile makes sense. 

Once the template gets Leaf support, we should just enable the necessary dockerfile line automatically.

Reply:
Yep, that can be a separate PR at that point.